external/llvm-project/mlir/lib/Transforms/Utils/OutlinerUtils.cpp

```diff
@@ -66,23 +72,15 @@ bool mlir::isConstantZero(Operation *op) {
 // Inspired by / adapted from outlineIfOp() in SCF/Transforms/Utils.cpp
 // and mergeIdenticalBlocks() in Utils/RegionUtils.cpp.
 
-struct OutliningCandidate {
-  OutliningCandidate(Operation *anchorOp, ArrayRef<Operation *> &trailingOps,
-                     ArrayRef<Operation *> &leadingOps, ArrayRef<Value> &params,
-                     ArrayRef<Value> &returnVals, StringRef partFnName);
+class OutliningCandidate {
+  SmallVector<Type, FUSION_CAPACITY_DEFAULT> params;
```

Comment:
Those'll be some pretty big stack allocations. Might as well leave these at their default sizes? Or drop the constant down to, say, 4 or 8?

Reply:
The benefit of a large initial allocation is that we delay allocating when adding to the vector.  Is the time taken for the realloc significant enough to worry about?